apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -84,6 +88,7 @@ private ApolloClient(Builder builder) {
     this.defaultHttpCacheControl = builder.defaultHttpCacheControl;
     this.defaultCacheControl = builder.defaultCacheControl;
     this.logger = builder.apolloLogger;
+    this.applicationInterceptors = builder.applicationInterceptors;
```

Comment:
pls wrap this into `Collections.unmodifiableList()` or even earlier in `ApolloClient.Builder`